packages/time-picker/src/vaadin-time-picker.js

```diff
@@ -631,6 +634,21 @@ class TimePicker extends PatternMixin(InputControlMixin(ThemableMixin(ElementMix
     }
   }
 
+  /** @private */
+  __updateAriaAttributes(items, opened, input) {
+    if (items === undefined || input === undefined) {
+      return;
+    }
+
+    if (items.length === 0) {
+      input.removeAttribute('role');
```

Comment:
Would combo-box also benefit from this fix?

Reply:
The combo-box case sounds more complicated to me. For example, it could be that with [custom filtering](https://vaadin.com/docs/latest/components/combo-box#custom-filtering) there would be 0 items, but as user is typing the updated filter could cause the dropdown items to become non-empty. Changing the `role` dynamically in this case could be problematic. So I'd prefer to limit this fix to time-picker for now.